### PR TITLE
fix: Use `@guardian/tsconfig`

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

Adds `@guardian/tsconfig` to the typscript config

## Why?

It's stricter (in a good way) and preps the code for import to https://github.com/guardian/csnx

### CODE REVIEW

**Review by ob6160 — DISMISSED** [review]
nice work 👍

File: src/cookies/getCookie.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export const getCookie = ({
 
 	const cookieVal = getCookieValues(name);
 
-	if (cookieVal.length > 0) {
+	if (cookieVal[0]) {
```
Comment (ob6160) [inline src/cookies/getCookie.ts]: i wonder if we could shift the array access up to `const cookieVal = getCookieValues(name)[0];` for consistency with https://github.com/guardian/libs/pull/390/files#diff-fed7a7bee596fcd23f90f5770a22a3d2828ce9d22c8a0f46518c6c6715798498R48 for example?

File: src/datetime/timeAgo.ts [hunk 2]
```diff
@@ -137,7 +105,9 @@ export const timeAgo = (
 		// Simple date - "9 Nov 2019"
 		return [
 			then.getDate(),
-			verbose ? longMonth(then.getMonth()) : shortMonth(then.getMonth()),
+			verbose
```
Comment (ob6160) [inline src/datetime/timeAgo.ts:108]: ah! a lot nicer :)

File: src/datetime/timeAgo.ts [hunk 3]
```diff
@@ -137,7 +105,9 @@ export const timeAgo = (
...
+				? then.toLocaleString('default', { month: 'long' })
+				: then.toLocaleString('default', { month: 'short' }),
```
Comment (sndrs) [inline src/datetime/timeAgo.ts:110]: @oliverlloyd how do you feel about these? guardian is at 98.99% support (greater than es module support). are they going to have wired side effects?

Comment (JamieB-gu) [reply src/datetime/timeAgo.ts:110]: Given that the other functions in this file return English strings, and the style guide is British English, is it possible we want to use `'en-GB'` here?

```suggestion
				? then.toLocaleString('en-GB', { month: 'long' })
				: then.toLocaleString('en-GB', { month: 'short' }),
```

I think using `'default'` will use the browser/system's default locale, and will therefore give the month in the language for that locale? For example, in French if the system is set to French. You can test this with the following snippet of code:

```js
const d = new Date();

d.toLocaleString('default', { month: 'long' });
```

in Node by setting the `LANG` environment variable:

```sh
LANG=fr_FR node
```

or in Safari by changing the macOS system language to e.g. French, restarting your browser, and then pulling up a console.

File: src/cookies/getCookie.ts [hunk 4]
```diff
@@ -18,13 +18,13 @@ export const getCookie = ({
 	const memoizedCookie = memoizedCookies.get(name);
 	if (memoizedCookie) return memoizedCookie;
 
-	const cookieVal = getCookieValues(name);
+	const value = getCookieValues(name)[0];
```
Comment (mxdvl) [inline src/cookies/getCookie.ts:21]:
```suggestion
	const [value] = getCookieValues(name);
```

File: src/cookies/setCookie.ts [hunk 5]
```diff
@@ -45,6 +45,9 @@ export const setCookie = ({
 
 	// If the cookie is already memoized we want to replace its value
 	if (memoizedCookies.has(name)) {
-		memoizedCookies.set(name, getCookieValues(name)[0]);
+		const value = getCookieValues(name)[0];
```
Comment (mxdvl) [inline src/cookies/setCookie.ts:48]:
```suggestion
		const [value] = getCookieValues(name);
```

File: src/cookies/setSessionCookie.ts [hunk 6]
```diff
@@ -26,6 +26,9 @@ export const setSessionCookie = ({
 
 	// If the cookie is already memoized we want to replace its value
 	if (memoizedCookies.has(name)) {
-		memoizedCookies.set(name, getCookieValues(name)[0]);
+		const value = getCookieValues(name)[0];
```
Comment (mxdvl) [inline src/cookies/setSessionCookie.ts:29]:
```suggestion
		const [value] = getCookieValues(name);
```

File: src/datetime/timeAgo.ts (hunk 3 above)
Comment (mxdvl) [reply src/datetime/timeAgo.ts:110]: I would be tempted to move this to a separate PR, as it’s possibly a breaking change, or at least most definitly a feature.

My suggestion for the existing code would be:

```ts
const longMonth = (month: number): string =>
	([
		'January',
		'February',
		'March',
		'April',
		'May',
		'June',
		'July',
		'August',
		'September',
		'October',
		'November',
		'December',
	] as const)[month] ?? "Unknown";
	
const shortMonth = (month: number) => longMonth(month).slice(0,3);
```

File: tsconfig.json [hunk 7]
```diff
@@ -1,18 +1,19 @@
...
-  "exclude": ["node_modules", "dist", "coverage"],
-  "ts-node": { "compilerOptions": { "module": "commonjs" } }
+	"extends": "@guardian/tsconfig",
+	"compilerOptions": {
+		"allowJs": true,
+		"allowSyntheticDefaultImports": true,
+		"esModuleInterop": true,
+		"forceConsistentCasingInFileNames": true,
+		"isolatedModules": true,
+		"module": "ESNext",
```
Comment (mxdvl) [inline tsconfig.json:9]: I think you can remove `allowSyntheticDefaultImports`, `esModuleInterop` and `module` and `moduleResolution`.

**Review by mxdvl — APPROVED** [review]
This is great, but I would not change the behaviour of `timeAgo` as part of this change.

File: src/datetime/timeAgo.ts (hunk 3 above)
Comment (sndrs) [reply src/datetime/timeAgo.ts:110]: this is a great point – `'en-GB'` should resolve that though?

**Comment by sndrs** [comment]
> I don’t think this should go in as-is, because of the timeAgo changes.

what do you think needs to change to make them ok? the `?? "Unknown"` is only there because as far as i know TS doesn't have a type for `0-11`. that makes it a branch of code we can't test, and given the month index comes from `getMonth`, a branch we'll never use. it's only there because `getMonth` returns a `number` in ts.

if the `en-GB` @JamieB-gu mentioned returns the same strings, shouldn't we be ok? what would you think we'd need to address in a subsequent PR?

File: src/datetime/timeAgo.ts (hunk 3 above)
Comment (oliverlloyd) [reply src/datetime/timeAgo.ts:110]: I think `then.toLocaleString('en-GB', { month: 'short' })` best respects the design guide

Comment (mxdvl) [reply src/datetime/timeAgo.ts:110]: I thought that [`toLocaleString` was not well supported in browsers](https://caniuse.com/mdn-javascript_builtins_date_tolocalestring), but it seems like that’s not the case, there’s only missing data for about 3% of browsers.

**Review by mxdvl — APPROVED** [review]
Indeed, with the great support of `toLocaleString`, I think this is good to go!

File: src/datetime/timeAgo.ts (hunk 3 above)
Comment (sndrs) [reply src/datetime/timeAgo.ts:110]: Support is available for everything that supports es modules afaict?

**Comment by JamieB-gu** [comment]
@sndrs did we want to change the `toLocaleString` call to specify `en-GB`? I think the code now on `main` will show datetimes in different languages if any of the consumers update to it? I may have missed something though 🤔

**Comment by sndrs** [comment]
Doh! Yes I completely forgot to implement it!

**Comment by sndrs** [comment]
@JamieB-gu correct locale went out in https://github.com/guardian/libs/releases/tag/v7.1.3

**Pull request merged**